lib/block.rb

```diff
@@ -0,0 +1,17 @@
+module Hotel
+	class Block 
+		attr_reader :id, :rooms, :rate, :date_range
+
+		def initialize(id:, rooms:, rate:, start_date:, end_date:)
+			@id = id
+			@rooms = rooms
+			@rate = rate
+			@date_range = Hotel::DateRange.new(
+				start_date: start_date, 
+				end_date: end_date
+			)
+			
+			raise ArgumentError.new("Block must contain between 2 and 5 rooms!") if @rooms.length < 2 || @rooms.length > 5
```

Comment:
I'd generally recommend raising errors like this before initializing fields.  It doesn't matter here but can be helpful as constructors get more complex.

Also, it's helpful to include the bad argument when you raise an `ArgumentError`:
```suggestion
			raise ArgumentError.new("Block must contain between 2 and 5 rooms! Got: #{@rooms}") if @rooms.length < 2 || @rooms.length > 5
```